pkg/controller/certificates/signer/signer.go

```diff
@@ -89,6 +89,19 @@ func newSigner(caFile, caKeyFile string, client clientset.Interface, certificate
 }
 
 func (s *signer) handle(csr *capi.CertificateSigningRequest) error {
+	// Special-case handling here in case we are talking to an older apiserver.
+	if csr.Spec.SignerName == nil {
+		return nil
```

Comment:
For nil, signerNames, I expected the same defaulting as in the apiserver, per the KEP

```go
csr, err := parse
switch {
  case err:
    legacy
  case iskubeletclient(csr):
    kubelet client
  case isclient(csr):
    client
  default:
    legacy
}
```